src/btree/bt_curnext.c

```diff
@@ -792,6 +795,22 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
 skip_page:
         if (F_ISSET(cbt, WT_CBT_READ_ONCE))
             LF_SET(WT_READ_WONT_NEED);
+
+#ifdef HAVE_DIAGNOSTIC
+        if (cbt->last_ref != NULL)
+            WT_ERR(__wt_page_release(session, cbt->last_ref, 0));
+
+        cbt->last_ref = cbt->ref;
+        if (cbt->last_ref) {
+            WT_ERR(__wt_hazard_set(session, cbt->last_ref, &busy));
+            // Loop until we get an hazard pointer.
+            while (busy) {
+                __wt_yield();
+                WT_STAT_CONN_INCR(session, page_busy_blocked);
+                WT_ERR(__wt_hazard_set(session, cbt->last_ref, &busy));
+            }
+        }
+#endif
```

Comment:
I don't like the idea of pinning two pages on every cursor as it walks the tree, that's going to significantly change how cursors affect the tree and especially how eviction and cursors interact. Specifically, this could cause diagnostic mode to hide bugs because cursors will pin two pages as we page-swap through the tree.

I realize it's `DIAGNOSTIC` only, but we run a lot of things in diagnostic mode, and I think it's a bad idea. (And, to be clear, I don't have any evidence this will affect anything, but my guess is that it will.)